### ISSUE TITLE
useReactiveVar unreliable with fast changing primitive

### ISSUE THREAD

**Issue by dontsave** [apollographql/apollo-client#7694]
i'm using a reactive var to track rollover state as a boolean, and I've noticed that `useReactiveVar` is unreliable if the state updates quickly (which is often the case with rollovers).

```tsx
const rVar = makeVar(true);

const Demo = () => {
  const val = useReactiveVar(rVar);
  console.log(val);

  return (
    <div
      style={{ marginTop: 200, border: `1px solid red` }}
      onMouseEnter={() => {
        console.log(`enter`, rVar());
        if (!rVar()) rVar(true);
      }}
      onMouseLeave={() => {
        console.log(`leave`, rVar());
        if (rVar()) rVar(false);
      }}
    >
      val: {val ? `true` : `false`}
    </div>
  );
};
```

In this simplified demo, if the user rolls over the div  quickly and repeatedly for awhile, the component will eventually no longer rerender, despite the boolean value in the reactive var updating. 

An apparent workaround is to use store the state in a non-primitive like `{val:true}` which seems to reliably trigger a rerender

**Versions**
ver 3.3.9

**Comment by benjamn** [comment]
@dontsave Can you try making this reproduction runnable, using [this template](https://github.com/apollographql/react-apollo-error-template/)? I tried, but I couldn't get the component to stop updating, using 3.3.9 or any later version.

Other questions that might be relevant:
1. What exact versions of `react` and `react-dom` are you using?
2. What browser version?

**Comment by felixmeziere** [comment]
Similar problem here, when I update a reactive var too often successively, at some point the re-rendering stops working :/

**Comment by felixmeziere** [comment]
Update: I can confirm that with `@apollo/client` v3.3.6 I don't have the issue, so 3.3.7 broke `useReactiveVar`.

**Comment by brainkim** [comment]
I am reasonably certain this is fixed by #8022. Feel free to ping and yell at me if this is not the case 🤗 Should be in a release soon.

**Comment by brainkim** [comment]
3.3.16
